Documentation/Plugins.md

```diff
@@ -203,13 +207,9 @@ struct MyBuildToolPlugin: BuildToolPlugin {
 }
 ```
 
-In the case of prebuild commands, any dependencies must be binary targets, since these commands run before the build starts.
-
-Note that a build tool plugin can return a combination of build tool commands and prebuild commands.  After the plugin runs, any build commands are incorporated into the build graph, which may result in changes that require commands to run during the subsequent build.
-
-Any prebuild commands are run after the plugin runs but before the build starts, and any files that are in the prebuild command's declared `outputFilesDirectory` will be evaluated as if they had been source files in the target.  The prebuild command should add or remove files in this directory to reflect the results of having run the command.
+A prebuild command has no inputs, so it will never be re-run due to changes in source files. The only trigger for re-running a prebuild command is a change to a declared dependendencies, which can only be (prebuilt) binary targets, since these commands run before any other targets have been built.
```

Comment:
@tomerd would we ever want to change this behavior? As currently specified it has consequences for https://github.com/apple/swift-package-manager/pull/7053, where we'll need to rerun these prebuild commands when target triple changes. We get the target triple when the build process is initiated and earlier, certainly not during resolution time as

> a change to a declared dependendencies

makes me think.

Reply:
IIUC this is just a question for @tomerd and doesn't have a bearing on the acceptability of this PR?

Reply:
I think that is def worth considering changing, such that there are more conditions which trigger re-running rebuild commands.

Reply:
IDK, specifying that "The only trigger for re-running a prebuild command is a change to declared dependencies" feels too strong to me if we have to change this behavior in the process of my work on https://github.com/apple/swift-package-manager/pull/7053

Reply:
I don't understand.  If you want to change the behavior of the system you can change the documentation appropriately.  Why the reluctance to precisely document the current behavior?

Reply:
agree with Dave the doc should reflect the state of things and then updated when behavior changes

Reply:
I agree as well — the documentation should match the implementation, and change when the behaviour changes.

Is it true today that today's implementation doesn't run prebuild commands at the start of every build?  That sounds different from the original intent.  IIRC, the only reason prebuild commands exist in the first place is because the two primary build systems for Swift Packages (`swift build` and `xcbuild`) don't support having a command generate an arbitrary set of output files based on the contents of its inputs.  So we needed to have some command that can always run at the start of every build to generate arbitrary output files, using caches to avoid unnecessary work.

Reply:
If that's the case I should obviously change the wording to reflect it… but that said I see no reason for them to run before every build if the only goal is to “generate files whose names can't be predicted from inputs.”  You can do that once and record a marker that it's been done, then use those files over and over until they are cleaned or declared dependencies of the prebuild command change.  If running before every build is necessary, there must be some other, currently unstated motivation for that requirement.  In fact, I extrapolated the behavior I documented from what I had been told about a prebuild command's reason for existence and the most efficient way to achieve that.

So that I document the right thing: are they run before *every* build of *every* target, or are they only run before builds of targets that are using the plugin?